pkg/chunked/compression_linux.go

```diff
@@ -157,10 +162,41 @@ func readEstargzChunkedManifest(blobStream ImageSourceSeekable, blobSize int64,
 	return manifestUncompressed, tocOffset, nil
 }
 
+func openTmpFile(tmpDir string) (fd int, err error) {
+	fd, err = unix.Open(tmpDir, unix.O_TMPFILE|unix.O_RDWR|unix.O_CLOEXEC, 0o600)
```

Comment:
Super minor but for this use case you can also add `O_EXCL` which signals that the file can never be linked (i.e. given a name later), in theory this allows optimizing things a bit.

Reply:
thanks.  Added now